packages/block-editor/src/components/inserter/media-tab/utils.js

```diff
@@ -27,10 +31,16 @@ export function getBlockAndPreviewFromMedia( media, mediaType ) {
 	const PreviewTag = mediaTypeTag[ mediaType ];
 	const preview = (
 		<PreviewTag
-			src={ mediaSrc }
+			src={ media.previewUrl || mediaSrc }
 			alt={ alt }
 			controls={ mediaType === 'audio' ? true : undefined }
 			inert="true"
+			onError={ ( { currentTarget } ) => {
+				// Fallback to the media source if the preview cannot be loaded.
+				if ( currentTarget.src === media.previewUrl ) {
+					currentTarget.src = mediaSrc;
```

Comment:
Any chance of runaway recursion here? Can the setting of `.src` to an invalid or inaccessible resource fire a new error event?

Reply:
Nope. We try two times. Once with the `previewUrl` and if that was used and failed, we try with the main src. If that fails we throw the error and it would mean 'bigger' problems with the external source..